numpy/core/records.py

```diff
@@ -661,6 +661,9 @@ def fromrecords(recList, dtype=None, shape=None, formats=None, names=None,
     """
 
     if formats is None and dtype is None:  # slower
+        if len(recList) == 0:
```

Comment:
I worry this does the wrong thing if `recList.shape = (0, ...)`

Reply:
Changed in 564db6f . I am not sure what is supposed to happen for different pathological cases. ``np.rec.fromrecords(np.empty((3, 0, 0)))`` still raises an error,  ``np.rec.fromrecords(np.empty((0, 3)))`` does not